source/particle/world.cc

```diff
@@ -535,6 +543,67 @@ namespace aspect
     }
 
     template <int dim>
+    void
+    World<dim>::exchange_ghost_particles()
+    {
+      TimerOutput::Scope timer_section(this->get_computing_timer(), "Particles: Exchange ghosts");
+
+      // First clear the current ghost_particle information
+      ghost_particles.clear();
+
+      std::multimap<types::subdomain_id, std::pair<types::LevelInd, Particle<dim> > > ghost_particles_by_domain;
+      std::vector<std::set<unsigned int> > vertex_to_neighbor_subdomain(this->get_triangulation().n_vertices());
+
+      typename DoFHandler<dim>::active_cell_iterator
+      cell = this->get_dof_handler().begin_active(),
+      endc = this->get_dof_handler().end();
+      for (; cell != endc; ++cell)
+        {
+          if (cell->is_ghost())
+            for (unsigned int v=0; v<GeometryInfo<dim>::vertices_per_cell; ++v)
+              vertex_to_neighbor_subdomain[cell->vertex_index(v)].insert(cell->subdomain_id());
```

Comment:
I suppose this code is the same as in deal.II's DoFHandler ghost exchange of DoF indices? I wonder if we ought to put this functionality somewhere...